inginious/frontend/plugins/grader_generator/pages/grader.py

```diff
@@ -65,3 +66,45 @@ def grader_generator_tab(course, taskid, task_data, template_helper):
 def grader_footer(course, taskid, task_data, template_helper):
     renderer = template_helper.get_custom_renderer(BASE_TEMPLATE_FOLDER, layout=False)
     return str(renderer.grader_templates()) + str(renderer.notebook_grader_test_form_modal())
+
+
+def on_file_deleted(course, task_id, path, task_factory):
+    """ It's called when a file of a task is deleted.
+    This function removes The test related with the file that was deleted if the file was a root file
+    in a multi lang environment
+     """
+    if not is_multi_lang(course, task_id):
+        return
+    file_name = get_file_name_from_path(path)
+    if file_name:
+        task_data = task_factory.get_task_descriptor_content(course.get_id(), task_id)
+        one_test_was_removed = remove_test_by_file_name(file_name, task_data)
+        if one_test_was_removed:
+            task_factory.update_task_descriptor_content(course.get_id(), task_id, task_data, "yaml")
+
+
+def is_multi_lang(course, task_id):
```

Comment:
```suggestion
def _is_allowed_environment(course, task_id):
```
Also, can you add a `_` at the beginning of the functions that are not used outside this file? That way we know which functions are imported by other files.